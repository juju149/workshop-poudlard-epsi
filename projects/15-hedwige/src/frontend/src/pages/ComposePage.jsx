import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import '../styles/ComposePage.css';

function ComposePage({ sessionId, user, onLogout }) {
  const [to, setTo] = useState('');
  const [subject, setSubject] = useState('');
  const [body, setBody] = useState('');
  const [sending, setSending] = useState(false);
  const [error, setError] = useState(null);
  const [success, setSuccess] = useState(false);
  const navigate = useNavigate();

  const handleSubmit = async (e) => {
    e.preventDefault();
    
    if (!to || !subject || !body) {
      setError('Please fill in all fields');
      return;
    }

    try {
      setSending(true);
      setError(null);
      setSuccess(false);

      const response = await fetch('/api/emails/send', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          'x-session-id': sessionId
        },
        body: JSON.stringify({ to, subject, body })
      });

      const data = await response.json();

      if (response.ok) {
        setSuccess(true);
        setTimeout(() => {
          navigate('/mailbox');
        }, 2000);
      } else {
        setError(data.error || 'Failed to send email');
      }
    } catch (err) {
      setError('Failed to send email');
    } finally {
      setSending(false);
    }
  };

  const handleCancel = () => {
    navigate('/mailbox');
  };

  return (
    <div className="compose-page">
      <header className="compose-header">
        <div className="header-left">
          <h1>ðŸ¦‰ Hedwige</h1>
        </div>
        <div className="header-right">
          {user && (
            <div className="user-info">
              {user.picture && <img src={user.picture} alt={user.name} />}
              <span>{user.name}</span>
            </div>
          )}
          <button onClick={onLogout} className="logout-btn">Logout</button>
        </div>
      </header>

      <div className="compose-content">
        <div className="compose-container">
          <h2>New Message</h2>

          {error && <div className="error-message">{error}</div>}
          {success && <div className="success-message">Email sent successfully!</div>}

          <form onSubmit={handleSubmit} className="compose-form">
            <div className="form-group">
              <label htmlFor="to">To:</label>
              <input
                type="email"
                id="to"
                value={to}
                onChange={(e) => setTo(e.target.value)}
                placeholder="recipient@example.com"
                disabled={sending}
              />
            </div>

            <div className="form-group">
              <label htmlFor="subject">Subject:</label>
              <input
                type="text"
                id="subject"
                value={subject}
                onChange={(e) => setSubject(e.target.value)}
                placeholder="Email subject"
                disabled={sending}
              />
            </div>

            <div className="form-group">
              <label htmlFor="body">Message:</label>
              <textarea
                id="body"
                value={body}
                onChange={(e) => setBody(e.target.value)}
                placeholder="Write your message here..."
                rows="15"
                disabled={sending}
              />
            </div>

            <div className="form-actions">
              <button 
                type="submit" 
                className="send-btn"
                disabled={sending}
              >
                {sending ? 'Sending...' : 'Send Email'}
              </button>
              <button 
                type="button" 
                className="cancel-btn"
                onClick={handleCancel}
                disabled={sending}
              >
                Cancel
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  );
}

export default ComposePage;
