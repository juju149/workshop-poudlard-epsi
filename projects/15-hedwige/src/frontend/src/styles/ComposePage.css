.compose-page {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  background: #f5f5f5;
}

.compose-header {
  background: white;
  padding: 15px 30px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.compose-content {
  flex: 1;
  display: flex;
  justify-content: center;
  padding: 40px 20px;
}

.compose-container {
  background: white;
  border-radius: 8px;
  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.1);
  padding: 30px;
  width: 100%;
  max-width: 800px;
}

.compose-container h2 {
  margin-bottom: 20px;
  color: #333;
}

.error-message {
  background: #fee;
  color: #c33;
  padding: 12px;
  border-radius: 6px;
  margin-bottom: 20px;
}

.success-message {
  background: #efe;
  color: #3c3;
  padding: 12px;
  border-radius: 6px;
  margin-bottom: 20px;
}

.compose-form {
  display: flex;
  flex-direction: column;
  gap: 20px;
}

.form-group {
  display: flex;
  flex-direction: column;
  gap: 8px;
}

.form-group label {
  font-weight: 500;
  color: #333;
}

.form-group input,
.form-group textarea {
  padding: 12px;
  border: 1px solid #ddd;
  border-radius: 6px;
  font-family: inherit;
  font-size: 1em;
  transition: border-color 0.3s;
}

.form-group input:focus,
.form-group textarea:focus {
  outline: none;
  border-color: #667eea;
}

.form-group textarea {
  resize: vertical;
  min-height: 200px;
}

.form-actions {
  display: flex;
  gap: 10px;
  margin-top: 10px;
}

.send-btn,
.cancel-btn {
  padding: 12px 24px;
  border: none;
  border-radius: 6px;
  font-weight: 500;
  font-size: 1em;
  transition: all 0.3s;
}

.send-btn {
  background: #667eea;
  color: white;
  flex: 1;
}

.send-btn:hover:not(:disabled) {
  background: #5568d3;
  transform: translateY(-1px);
  box-shadow: 0 4px 8px rgba(102, 126, 234, 0.3);
}

.cancel-btn {
  background: #f0f0f0;
  color: #333;
}

.cancel-btn:hover:not(:disabled) {
  background: #e0e0e0;
}
