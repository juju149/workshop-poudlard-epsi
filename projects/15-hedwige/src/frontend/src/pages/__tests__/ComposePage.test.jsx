import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import ComposePage from '../ComposePage';

describe('ComposePage', () => {
  it('correspond au snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <ComposePage />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
