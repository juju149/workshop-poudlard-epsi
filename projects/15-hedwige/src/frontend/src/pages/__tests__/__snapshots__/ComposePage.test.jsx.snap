// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ComposePage > correspond au snapshot 1`] = `
<DocumentFragment>
  <div
    class="compose-page"
  >
    <header
      class="compose-header"
    >
      <div
        class="header-left"
      >
        <h1>
          ðŸ¦‰ Hedwige
        </h1>
      </div>
    </header>
    <div
      class="compose-content"
    >
      <div
        class="compose-container"
      >
        <h2>
          New Message
        </h2>
        <form
          class="compose-form"
        >
          <div
            class="form-group"
          >
            <label
              for="to"
            >
              To:
            </label>
            <input
              id="to"
              placeholder="recipient@example.com"
              type="email"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="subject"
            >
              Subject:
            </label>
            <input
              id="subject"
              placeholder="Email subject"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="body"
            >
              Message:
            </label>
            <textarea
              id="body"
              placeholder="Write your message here..."
              rows="15"
            />
          </div>
          <div
            class="form-actions"
          >
            <button
              class="send-btn"
              type="submit"
            >
              Send Email
            </button>
            <button
              class="cancel-btn"
              type="button"
            >
              Cancel
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
